src/steps/active-directory/converters.ts

```diff
@@ -0,0 +1,227 @@
+        name: data.displayName,
+        id: data.id,
+        category: 'endpoint',
+        fqdn: data.computerDnsName as string | undefined,
+        hostname: data.computerDnsName as string | undefined,
+        osName: data.osPlatform as string | undefined,
+        osVersion: data.osVersion as string | undefined,
+        platform: data.osPlatform as string | undefined,
+      },
+      skipTargetCreation: true,
+    },
+  });
+}
+
+export function generateUserKey(user: User): string {
+  return user.id as string;
+}
+
+export function createUserEntity(data: UserLogon): Entity {
+  return createIntegrationEntity({
+    entityData: {
+      source: {}, // removed due to size
+      assign: {
+        _key: generateUserKey(data),
+        _class: entities.USER._class,
+        _type: entities.USER._type,
+        id: data.id,
+        name: data.accountName,
+        domain: data.accountDomain,
+        username: data.accountName,
+        displayName: data.accountName as string | undefined,
+        firstSeen: data.firstSeen,
+        lastSeen: data.lastSeen,
+        logonTypes: data.logonTypes,
+        active: data.isDomainAdmin,
+      },
+    },
+  });
+}
+
+export function createFindingEntity(defenderFinding: Finding): Entity {
+  return createIntegrationEntity({
+    entityData: {
+      source: {},
+      assign: {
+        id: defenderFinding.id,
+        _type: entities.FINDING._type,
+        _class: entities.FINDING._class,
+        _key: defenderFinding.id + '_' + Math.random() * 1000000,
+        name: defenderFinding.name,
+        displayName: defenderFinding.name,
+        description: defenderFinding.description,
+        severity: defenderFinding.severity,
+        score: defenderFinding.cvssV3,
+        numericSeverity: 0,
+        publishedOn: defenderFinding.publishedOn,
+        updatedOn: getTime(defenderFinding.updatedOn),
+        category: '',
+        open: true,
+        references: defenderFinding.exploitUris,
+        exposedMachines: defenderFinding.exposedMachines,
+      },
+    },
+  });
+}
+
+export function createMachineUserRelationship(
+  machine: Entity,
+  user: Entity,
+): Relationship {
+  return createDirectRelationship({
+    _class: RelationshipClass.HAS,
+    from: machine,
+    to: user,
+  });
+}
+
+export function createAccountMachineRelationship(
+  account: Entity,
+  machine: Entity,
+): Relationship {
+  return createDirectRelationship({
+    _class: RelationshipClass.HAS,
+    from: account,
+    to: machine,
+  });
+}
+
+export function createFindingsCveRelationship(
+  findingEntity: Entity,
+): Relationship {
+  return createMappedRelationship({
+    _class: RelationshipClass.IS,
+    _type: MappedRelationships.FINDING_IS_CVE_VULNERABILITY._type,
+    _mapping: {
+      sourceEntityKey: findingEntity._key,
+      relationshipDirection:
+        MappedRelationships.FINDING_IS_CVE_VULNERABILITY.direction,
+      targetFilterKeys: [['Resource', '_type', '_class', '_key']],
```

Comment:
```suggestion
      targetFilterKeys: [['_type', '_key']],
```
This is all that is necessary in the case of connecting the `Finding` to a `_type: 'cve'` once the `_key` is properly set to the CVE identifier.